pkg/ccl/backupccl/restore_job.go

```diff
@@ -474,7 +476,9 @@ func restore(
 				requestFinishedCh,
 				tracingAggCh,
 				genSpan,
-			), "sending remote AddSSTable requests")
+			)
+			onlineRestoreStats <- roachpb.RowCount{Rows: approxRows, DataSize: approxDataSize}
```

Comment:
nit: since the goroutine is guaranteed to finish before `restore()` returns, you don't need to pass all of this through a channel. At the end of the `sendAddRemoteSSTs` goroutine you could set `progressTracker.mu.res`. i.e:

```
linkRes, err := sendAddRemoteSSTs()
progressTracker.mu.Lock()
defer progressTracker.mu.Unlock()

//  During the link phase of online restore, we do not update stats progress as job occurs. We merely reuse the 
// `progressTracker.mu.res` var to reduce the number of local vars floating around in`restore`. 
progressTracker.mu.res = linkRes
```